oxidation/libparsec/crates/core_fs/src/storage/chunk_storage.rs

```diff
@@ -0,0 +1,409 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use diesel::dsl::count_star;
+use diesel::{sql_query, ExpressionMethods, Insertable, QueryDsl, RunQueryDsl};
+use std::sync::Mutex;
+
+use parsec_api_crypto::SecretKey;
+use parsec_api_types::{ChunkID, DateTime, DEFAULT_BLOCK_SIZE};
+
+use super::local_database::{SqliteConn, SQLITE_MAX_VARIABLE_NUMBER};
+use crate::error::{FSError, FSResult};
+use crate::extensions::CoalesceTotalSize;
+use crate::schema::chunks;
+
+#[derive(Insertable, AsChangeset)]
+#[diesel(table_name = chunks)]
+struct NewChunk<'a> {
+    pub chunk_id: &'a [u8],
+    pub size: i32,
```

Comment:
[integer as stored as signed, up to 64bits](https://www.sqlite.org/datatype3.html), so this should be `i64` in theory.

However I think it would be better if we set it as unsigned, and even better as `NonZeroU64` (which is not strictly true, it should be more like `NonZeroU63` but I don't think Rust support this type and we should be pretty safe from hitting such upper limit)

Reply:
Another approach is to stick with `NonZeroU32` which sqlite can store (unlike u64), and given chunk size is much murch smaller (typically ~512ko vs 2³²)

@vxgmichel @FirelightFlagboy what do you think ?

Reply:
I'm in favor of using unsigned type when the value don't make any sense when negative

Reply:
Yes we can iterate, could you create the issue ?

Reply:
It's not something you will use, it's an interface with Sqlite

Reply:
I warn that even if we have an unsigned integer, the representation will be signed so a little dangerous

Reply:
The integer size an signed/unsigned is a typical recipe for errors, we must do some thinking about it (and and comments to explain the rational)

But for the moment you can:
- add `TODO: think about type compat with sqlite`
- create an issue about that is the oxidation project